pkg/controller/migstage/migstage_controller.go

```diff
@@ -62,7 +66,10 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	err = c.Watch(
 		&source.Kind{Type: &migapi.MigPlan{}},
 		&handler.EnqueueRequestsFromMapFunc{
-			ToRequests: handler.ToRequestsFunc(MigPlanToMigStage),
+			ToRequests: handler.ToRequestsFunc(
+				func(a handler.MapObject) []reconcile.Request {
+					return migref.GetRequests(a, migapi.MigStage{})
```

Comment:
This is a nice enhancement! Like you say allows us to avoid having a function for each type combo.